### PR TITLE
feat(rust, python): Implement forward strategy in groupby join_asof

### PR DESCRIPTION
**Pull request by AlecZorab**

### CODE REVIEW

File: polars/polars-core/src/frame/asof_join/groups.rs [hunk 1]
```diff
@@ -730,4 +801,143 @@ mod test {
...
+
+        let out = trades.join_asof_by(
+            &quotes,
+            "time",
+            "time",
+            ["ticker"],
+            ["ticker"],
+            AsofStrategy::Forward,
+            None,
+        )?;
```
Comment (AlecZorab) [inline polars/polars-core/src/frame/asof_join/groups.rs:914]: I'm not sure why you'd ever be looking for quotes from after the trades, but who said that test cases have to make sense in terms of business logic? 🤷

Comment (ritchie46) [reply polars/polars-core/src/frame/asof_join/groups.rs:914]: I don't understand financial business, so to me it seems like a perfectly valid test. ;)

**Comment by ritchie46** [comment]
Awesome! Can you add the strategy to the python api as well?

**Comment by AlecZorab** [comment]
Was already plumbed through, so I just added some python tests

**Comment by ritchie46** [comment]
Thanks a lot @AlecZorab!

**Pull request merged**